### PR TITLE
Refector: Dynamically render products using JS Array and <template> Tag

### PR DESCRIPTION
**Pull request by manoelmndf**
### What was done:
- Replaced code of products in the html with a dynamic rendering solution
- Used the <template> in html tag
- JS separated from file html

[Issues #6]

### CODE REVIEW

File: appointment.html
Comment (ndvo) [inline appointment.html:1]: This code is not related to issue #6 . It is related to issue #7 .

Is this change intended to be on this PR?

File: petshop.html [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Products</title>
+    <link rel="stylesheet" href="assets/css/style.css">
+    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.0/dist/css/bootstrap.min.css" rel="stylesheet">
+    <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.3.0/dist/js/bootstrap.bundle.min.js"></script>
+    <script src="assets/js/loadcomponents.js"></script>
+    <script src="assets/js/product.js" defer></script>
+</head>
+
+<body onload="loadHeader()">
```
Comment (ndvo) [inline petshop.html:15]: this "loadHeader" function does not exist in the codebase.

File: appointment.html [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+
+    <div id="calendar-page" >
+        <div class="calendar-header">
+            <button id="prev-month">Previous</button>
+            <div id="current-month"></div>
+            <button id="next-month">Next</button>
+        </div>
+        <div class="calendar">
+            <!-- Dias da semana -->
```
Comment (ndvo) [inline appointment.html:26]:
```suggestion
```

We need to avoid unnecessary comments. No comments that can be easily derived from code.
Also, no Portuguese.

File: assets/css/calendar.css [hunk 3]
```diff
@@ -0,0 +1,29 @@
+/* Adicione estilo para o calendário e botões */
```
Comment (ndvo) [inline assets/css/calendar.css:1]:
```suggestion
```

File: teste.html [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        // Script para carregar o conteúdo do header
+        document.addEventListener("DOMContentLoaded", function () {
+            fetch("components/header.html")
+                .then(response => {
+                    //console.log("log do carregamento do arquivo",response);
+                    if (!response.ok) throw new Error("Não foi possível carregar o header.");
+                    return response.text();
+                })
+                .then(data => {
+                    //console.log("log data",data);
```
Comment (ndvo) [inline teste.html:20]:
```suggestion
```

File: teste.html [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+                    document.getElementById("header-placeholder").innerHTML = data;
+                })
+                .catch(error => console.error("Erro ao carregar o header:", error));
```
Comment (ndvo) [inline teste.html:23]: This message should be in English.

File: teste.html (hunk 4 above)
Comment (manoelmndf) [reply teste.html]: > 
teste.html file deleted.

File: teste.html [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+        });
+    </script>
+</head>
+
+<body>
+    <div id="header-placeholder"></div>
+    <main>
+        <!-- Conteúdo específico da página -->
```
[the remark this answers is not in the document]
Comment (manoelmndf) [reply teste.html]: teste.html file deleted.

**Review by ndvo — APPROVED** [review]
LGTM
